lib/BitrateTest.ts

```diff
@@ -172,20 +185,31 @@ export class BitrateTest extends EventEmitter {
       this._setupDataChannel();
       this._startTest();
     });
+
+    this._timeout = setTimeout(() => {
+      const error: TimeoutError = new TimeoutError(
+        'Network timeout, exceeded limit of `${BITRATE_TEST_TIMEOUT_MS}`ms');
+      this._errors.push(error);
+      this.emit(BitrateTest.Events.Error, error);
+      this.stop();
+    }, BITRATE_TEST_TIMEOUT_MS);
   }
 
   /**
    * Stops the current test.
    */
   stop(): void {
+    clearTimeout(this._timeout!);
     clearInterval(this._sendDataIntervalId!);
     clearInterval(this._checkBitrateIntervalId!);
 
-    this._pcSender.close();
-    this._pcReceiver.close();
-    this._endTime = Date.now();
+    if (typeof this._endTime !== 'number' || this._endTime === 0) {
```

Comment:
Ahhh. Just to double check, we wrapped this to an `if` statement so that calling `.stop()` multiple times doesn't set the endTime and doesn't emit `.on('end')` multiple times. Is this the only reason, or did you find other issues?

Reply:
That's right. I think a more robust solution here would be to implement a state for this test. Not sure if we've discussed that already?

Reply:
Or even, all tests. Even if it's just "running", "ended"

Reply:
Ok cool.
Yeah, states would be more robust. We haven't discussed that yet.